src/interpreter.rs

```diff
@@ -262,9 +475,7 @@ pub fn eval_script(
                     stack.push_back(vch_push_value.clone());
                 }
             }
-            Operation::Constant(value) => {
-                todo!()
-            }
+            Operation::Constant(value) => stack.push_back(vec![value as u8]),
```

Comment:
Should the payload of `Operation::Constant` just be a `u8`? The `as` cast here requires non-local reasoning.

Reply:
For this PR, I would really recommend that you do a side-by-side comparison of the full PR against the C++ code, rather than commit-by-commit. E.g., interpreter.rs should match interpreter.h followed by interpreter.cpp as closely as possible.

You’ll see that there’s not even an `Operation::Constant` in the later stages of the PR, and `as` is almost completely gone as well (with the remaining uses commented). And the changes to use `stack.pop()?` have happened as well.

Not that the history isn’t useful, but it’s mostly a path from Sean’s WIP to the full implementation, and has intermediate goals (like, “get the existing sparse C++ tests passing on the Rust impl”) that don’t really matter for the review.